worlds/hk/__init__.py

```diff
@@ -277,30 +277,34 @@ def _add(item_name: str, location_name: str):
 
         for option_key, option in hollow_knight_randomize_options.items():
             randomized = getattr(self.multiworld, option_key)[self.player]
-            for item_name, location_name in zip(option.items, option.locations):
-                if item_name in junk_replace:
-                    item_name = self.get_filler_item_name()
-
-                if (item_name == "Crystal_Heart" and self.multiworld.SplitCrystalHeart[self.player]) or \
-                        (item_name == "Mothwing_Cloak" and self.multiworld.SplitMothwingCloak[self.player]):
-                    _add("Left_" + item_name, location_name)
-                    _add("Right_" + item_name, "Split_" + location_name)
-                    continue
-                if item_name == "Mantis_Claw" and self.multiworld.SplitMantisClaw[self.player]:
-                    _add("Left_" + item_name, "Left_" + location_name)
-                    _add("Right_" + item_name, "Right_" + location_name)
-                    continue
-                if item_name == "Shade_Cloak" and self.multiworld.SplitMothwingCloak[self.player]:
-                    if self.multiworld.random.randint(0, 1):
-                        item_name = "Left_Mothwing_Cloak"
-                    else:
-                        item_name = "Right_Mothwing_Cloak"
-
-                _add(item_name, location_name)
+            if randomized or option_key not in logicless_options:
```

Comment:
not a big fan of an extra layer of indent somewhere that is already pretty nested. Could you extract some methods?

Reply:
since nothing else is needed in the parent for loop, would changing it to something like this (as to not change existing indents) feel better?
```
            if all([not randomized, option_key in logicless_options, not self.multiworld.SharedItems[self.player]]):
                continue
```

Reply:
Yeah I'd be fine with inverting the if as a solution.
Not sure about using all() instead of just an inverted condition, but that's subjective style so you do what you like

Reply:
I just `not`'d it at first,
```
if not(randomized or option_key not in logicless_options or self.multiworld.SharedItems[self.player]):
```
but i felt like it reads horribly and is hard to intuit what it's actually testing, if there's no optimization difference (and no other suggestions that read a bit nicer) I'll stick with the all() one